### DISCUSSION TITLE
Will the MLX design incorporate any of the stuff from this paper?

### DISCUSSION THREAD

**Discussion by dougdew64** [ml-explore/mlx#238] [General]
This stuff is so incredibly cool: https://arxiv.org/pdf/2312.11514.pdf

I seriously hope that MLX will evolve to leverage the kind of stuff which is mentioned in that paper.

It sure seems to me that Apple's vertically integrated stack is going to enable Apple to accomplish things which nobody else can accomplish.

**Comment by awni** [comment]
I’d love to see someone try to implement those techniques in MLX! I’m not sure if we have all the tools in the API needed for this, but would also be good to know what we are missing.
